submissions/dana-channel-routing/main.md

```diff
@@ -0,0 +1,138 @@
+* Less is more: if the same amount of information can be conveyed with four lines instead of twenty, that's probably a good thing to do.
+
+So now, it's time to *search* for better ideas. And the best way to research for anything, as we all know, is ~~google~~ by pressing "T" in Factorio:
+
+![Factorio tech tree](factorio-tech-tree.png)
+
+Factorio's graph renderer has a more subtle way of rendering links. Each link is decomposed into three segments: two verticals, one horizontal. This approach scales much better with wider graphs, because:
+
+* There's never a crossing of nearly parallel lines. All crossings are at right angle, which is optimal to prevent users from following the wrong line.
+* The density of links is kept under control: there's always enough gap between parallel lines to distinguish them from each other.
+
+The price for this readability is vertical space: there need to be enough room between the two rows of technologies to add all the horizontal segments without collisions:
+
+![Factorio tech tree](factorio-tech-tree-spacing.png)
+
+To minimize this cost, there's a simple yet huge optimisation: What if lines didn't link just two elements, but any number of elements? Just draw a single wide horizontal line for each item/technology, then add as many vertical lines as necessary to connect to the nodes.
+
+![Factorio tech tree dana link types](factorio-tech-tree-dana-link-types.png)
+
+Much more compact, less clutter, definitely *nice™ and understandable™*. This gives a "main bus" vibe to these graph sections that'll hopefully feel natural to a Factorio player, while hitting a good compromise between the general guidelines. This is also technically possible with Factorio's in-game rendering API, as links are just a bunch of lines, triangles and circles. This almost allows Dana to fit Factorio's full crafting graph on a single screen:
+
+![Dana: Factorio full graph](dana-full-graph.png)
+
+### The Coding Part
+
+So that's all for the drawing board, time to throw some code at the problem to get the job done! But with that come some other problems. First, Factorio Mods are made in a language called Lua, and Lua has an ridiculously barren ecosystem. No success in finding a library that can do this kind of link routing.
+
+Another solution would be porting a library from other languages. Sadly there are plenty of libraries to draw conventional graphs, but Dana is now dealing with links between any number of nodes. That's not a graph anymore, but an hypergraph. While the word sounds definitely cooler, there aren't many software libraries to draw them, and in general there's much less scientific literature on the topic. No success in finding clues to do routing the Dana way.
+
+So, Dana has a router made "almost" from scratch. "Almost" because there was a lot of inspiration to be found elsewhere, it just required looking into some unexpected places...
```

Comment:
```suggestion
So, Dana has a router made "almost" from scratch. "Almost", because there was a lot of inspiration to be found elsewhere, it just required looking into some unexpected places...
```